src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -221,5 +229,131 @@ public TElement[] ToArray()
         {
             return _source.ToArray(_minIndex, _maxIndex);
         }
+
+        public List<TElement> ToList()
+        {
+            return _source.ToList(_minIndex, _maxIndex);
+        }
+
+        public int GetCount(bool onlyIfCheap)
+        {
+            return _source.GetCount(_minIndex, _maxIndex, onlyIfCheap);
+        }
+    }
+
+    public static partial class Enumerable
```

Comment:
Simply by following where stuff had been before. That's meant most things have been inside, since they added to or replaced something that was inside. Those related to ordering ended up outside because `OrderedEnumerable` is.

That's followed more from keeping changes local to what they were about in what had been a massive file, than any conscious plan.

I'm inclined to have anything that derives from `Iterator` be inside, anything else outside.